src/react/components/pages/train/trainPage.tsx

```diff
@@ -120,6 +122,9 @@ export default class TrainPage extends React.Component<ITrainPageProps, ITrainPa
         this.checkAndUpdateInputsInLocalStorage(this.props.project.id);
         this.appInsights = getAppInsights();
         document.title = strings.train.title + " - " + strings.appName;
+        if (isElectron && this.props.project.sourceConnection.providerType === "localFileSystemProxy") {
+            this.setState({ isOnPrem: true });
```

Comment:
I don't think we need to add a state for this. There is no reason to re-render the page (by updating the state) if it's just used as a condition

Reply:
You're right, I'll fix it.